packages/magento-product/components/AddProductsToCart/useAddProductsToCartAction.ts

```diff
@@ -32,6 +36,25 @@ export function useAddProductsToCartAction(
     loading,
   } = props
 
+  const [showSuccess, setShowSuccess] = useState<boolean>(false)
+
+  const userErrors = toUserErrors(data)
+
+  const submitSuccesful =
+    !formState.isSubmitting && formState.isSubmitSuccessful && !error?.message && !userErrors.length
+
+  useEffect(() => {
+    if (submitSuccesful && submittedVariables?.cartItems.find((item) => item.sku === sku)) {
+      setShowSuccess(true)
+    }
+  }, [sku, submitSuccesful, submittedVariables?.cartItems])
+
+  if (showSuccess) {
+    setTimeout(() => {
+      setShowSuccess(false)
+    }, 2000)
```

Comment:
We should probably cancel this timeout when someone adds the same product multiple times before the timeout ends. Otherwise the checkmark will disappear too fast for the subsequent times the product is added.